src/teleop_twist_keyboard.py

```diff
@@ -90,31 +184,29 @@ def vels(speed,turn):
 				speed = speed * speedBindings[key][0]
 				turn = turn * speedBindings[key][1]
 
-				print(vels(speed,turn))
+				print(vels(speed, turn))
 				if (status == 14):
 					print(msg)
 				status = (status + 1) % 15
+				continue
 			else:
+				# Skip updating cmd_vel if key timeout and robot already
+				# stopped.
+				if key == '' and x == 0 and y == 0 and th == 0:
```

Comment:
The check of `x`, `y`, `th` seems incorrect. I think `pub_thread.x`, `pub_thread.y` and `pub_thread.th` should be checked instead

Reply:
I just copied the code from the original ROS repo. But I think these will always be the same when ending up at this check.